web/src/main/react-dashboard/src/components/TestRunView.js

```diff
@@ -196,12 +200,31 @@ class TestRunView extends Component {
       })
   }
 
+  async getgrafanaUrl(testid){
+    const grafanaUrl = TESTGRID_CONTEXT + '/api/test-plans/perf-url/' + testid;
+    const fetchResult = fetch(grafanaUrl, {
+                            method: "GET",
+                            credentials: 'same-origin',
+                            headers: {
+                            'Accept': 'application/json'
+                            }
+                       }).then(this.handleError);
+     const response = await fetchResult;
+     const urldata = await response.text();
+     console.log("new" + urldata);
+     this.setState({grafanaUrl: urldata});
+
+  }
+
+
   render() {
+    var pageURL = window.location.href;
+    const PERFDASH_URL = this.state.grafanaUrl
     const divider = (<Divider inset={false} style={{borderBottomWidth: 1}}/>);
-    const logAllContentUrl = TESTGRID_CONTEXT + '/api/test-plans/log/' +
-      window.location.href.split("/").pop() + "?truncate=" + false;
-    const turncatedRunLogUrl = TESTGRID_CONTEXT + '/api/test-plans/log/' +
-      window.location.href.split("/").pop() + "?truncate=" + true;
+    const logUrl = TESTGRID_CONTEXT + '/api/test-plans/log/' + pageURL.split("/").pop()
+    const logAllContentUrl = logUrl + "?truncate=" + false;
+    const turncatedRunLogUrl = logUrl + "?truncate=" + true;
+
```

Comment:
You could use,
`const logUrl = TESTGRID_CONTEXT + '/api/test-plans/log/' + pageURL.split("/").pop() + "?truncate="
const logAllContentUrl = logUrl + false;
const turncatedRunLogUrl = logUrl + true;`